### PR TITLE
feat(openapi): Add access key authentication in openapi check token.

### PR DESCRIPTION
**Pull request by shuofan**
#### What this PR does / why we need it:
Add access key authentication in openapi check token.

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMTc4LDExNzQsMTE5MCwxMjE4XSwiYXNzaWduZWUiOlsiMTAwMTA3MyJdfQ%3D%3D&id=306726&iterationID=1190&pId=306714&type=TASK)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  Add access key authentication in openapi check token.            |
| 🇨🇳 中文    |   openapi 增加 access key token 认证           |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/core/openapi-ng/auth/token/check.go [hunk 1]
```diff
@@ -64,12 +64,19 @@ func (p *provider) checkToken(opts openapiauth.Options, req *http.Request, token
 	}
 	// 2. openapi oauth2 token
 	oauthToken, err := auth.VerifyOpenapiOAuth2Token(p.oauth2server, &OAuth2APISpec{opts}, req)
+	if err == nil {
+		return clientToken{
+			ClientID:   oauthToken.ClientID,
+			ClientName: oauthToken.ClientName,
+		}, nil
+	}
+	accesskey, err := auth.VerifyAccessKey(p.TokenService, req)
```
Comment (sfwn) [inline modules/core/openapi-ng/auth/token/check.go:73]: add comment

File: modules/openapi/auth/token.go [hunk 2]
```diff
@@ -187,3 +189,30 @@ func matchAPISpec(accessibleAPI apistructs.AccessibleAPI, spec OAuth2APISpec) bo
...
 		strutil.Equal(accessibleAPI.Schema, spec.Scheme(), true)
 }
+
+func VerifyAccessKey(tokenService tokenpb.TokenServiceServer, r *http.Request) (TokenClient, error) {
+	auth := r.Header.Get(HeaderAuthorization)
+	token := ""
+	if auth != "" && strings.HasPrefix(auth, HeaderAuthorizationBearerPrefix) {
+		token = auth[len(HeaderAuthorizationBearerPrefix):]
+	}
+	resp, err := tokenService.QueryTokens(context.Background(), &tokenpb.QueryTokensRequest{
```
Comment (sfwn) [inline modules/openapi/auth/token.go:199]: r.ctx

File: modules/openapi/auth/token.go [hunk 3]
```diff
@@ -187,3 +189,30 @@ func matchAPISpec(accessibleAPI apistructs.AccessibleAPI, spec OAuth2APISpec) bo
...
+	if err != nil || resp == nil {
+		return TokenClient{}, err
+	}
+	if resp.Total == 0 {
+		return TokenClient{}, fmt.Errorf("auth failed, access key: %s", token)
+	} else if resp.Total > 1 {
+		return TokenClient{}, fmt.Errorf("auth failed, duplidate data")
+	}
+	if resp.Data[0].ScopeId != "" {
+		r.Header.Set(httputil.InternalHeader, resp.Data[0].ScopeId)
```
Comment (sfwn) [inline modules/openapi/auth/token.go:213]: set token client id and name

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/openapi/auth/token.go (hunk 2 above)
Comment (shuofan) [reply modules/openapi/auth/token.go]: done

File: modules/core/openapi-ng/auth/token/check.go (hunk 1 above)
Comment (shuofan) [reply modules/core/openapi-ng/auth/token/check.go:75]: done

**Review by sfwn — APPROVED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**